### ISSUE TITLE
Add column name repairing for all IO sources as a common step for dataframe creation

### ISSUE THREAD

**Issue by zaleslaw** [Kotlin/dataframe#387]
The solution could be based on the PR #386

**Comment by zaleslaw** [comment]
Probably, default should be changed and more tests should be added https://github.com/Kotlin/dataframe/discussions/414

**Comment by koperagen** [comment]
FYI, there's org.jetbrains.kotlinx.dataframe.impl.ColumnNameGenerator that is used a lot for creating unique names
I added public API on top of it org.jetbrains.kotlinx.dataframe.api.DynamicDataFrameBuilder

**Comment by Jolanrensen** [comment]
Indeed, we should use ColumnNameGenerator to be consistent across DataFrame. JDBC and Excel both have their own solution now, which is not ideal at all
